core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -81,7 +81,7 @@ public static String getFolderContainingFile(String carbonFilePath) {
     int lastIndex = carbonFilePath.lastIndexOf('/');
     // below code for handling windows environment
     if (-1 == lastIndex) {
-      lastIndex = carbonFilePath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR);
+      lastIndex = carbonFilePath.lastIndexOf(CarbonCommonConstants.WINDOWS_FILE_SEPARATOR);
```

Comment:
why it need to use windows file separator?

Reply:
carbonFilePath is in type of '/root/schema', while it is in type of 'D:\schema'. The separator is different, dividing into FILE_SEPARATOR ('/') and WINDOWS_FILE_SEPARATOR ('\'). 

In the Windows Env.  The variant 'lastIndex' is always -1, carbonFilePath.substring(0, lastIndex) will be throw SubStringIndexOutOfBound Exception.

Reply:
in the linux env, is lastIndex right if use 'WINDOWS_FILE_SEPARATOR'?
it better set the right separator according to system env, for example: use java.io.File.separator

Reply:
Yes, java.io.File.separator is more elegant. I have modified code to use java.io.File.separator as you suggested.